tests/Unit/DataStructures/Test_DenseMatrix.cpp

```diff
@@ -0,0 +1,37 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "tests/Unit/TestingFramework.hpp"
+
+#include "DataStructures/DenseMatrix.hpp"
+#include "tests/Unit/TestHelpers.hpp"
+
+SPECTRE_TEST_CASE("Unit.DataStructures.DenseMatrix", "[DataStructures][Unit]") {
+  // Since `DenseMatrix` is just a thin wrapper around `blaze::DynamicMatrix` we
+  // test a few operations and refer to Blaze for more thorough tests.
+  DenseMatrix<double> matrix(2UL, 3UL, 1.);
+  CHECK(matrix.rows() == 2);
+  CHECK(matrix.columns() == 3);
+  CHECK(matrix(0, 0) == 1.);
+  CHECK(matrix(0, 1) == 1.);
+  CHECK(matrix(0, 2) == 1.);
+  CHECK(matrix(1, 0) == 1.);
+  CHECK(matrix(1, 1) == 1.);
+  CHECK(matrix(1, 2) == 1.);
+  DenseMatrix<int> A{{2, 1}, {0, 1}};
+  DenseMatrix<int> B{{1, 0}, {3, 2}};
+  CHECK(A + B == DenseMatrix<int>{{3, 1}, {3, 3}});
+  CHECK(2 * A == DenseMatrix<int>{{4, 2}, {0, 2}});
+  CHECK(A * B == DenseMatrix<int>{{5, 2}, {3, 2}});
+  CHECK(B * A == DenseMatrix<int>{{2, 1}, {6, 5}});
+
+  test_copy_semantics(matrix);
+  auto matrix_copy = matrix;
+  test_move_semantics(std::move(matrix), matrix_copy);
+}
+
+SPECTRE_TEST_CASE("Unit.Serialization.DenseMatrix",
```

Comment:
I think we've been moving away from separate serialization tests (as opposed to just merging them into the main test).  I think there used to be some technical reason they needed to be separate that has been solved, although I can't remember what it was anymore.